lightning-dns-resolver/src/lib.rs

```diff
@@ -458,8 +458,8 @@ mod test {
 		}
 
 		check_added_monitors(&nodes[1], 1);
-		let updates = get_htlc_update_msgs!(nodes[1], payer_id);
-		nodes[0].node.handle_update_fulfill_htlc(payee_id, &updates.update_fulfill_htlcs[0]);
+		let mut updates = get_htlc_update_msgs!(nodes[1], payer_id);
+		nodes[0].node.handle_update_fulfill_htlc(payee_id, updates.update_fulfill_htlcs.remove(0));
```

Comment:
I am not sure if I like the mutable list just to get the first one. I get it that it needs to be moved now.